tree-sitter-stack-graphs/src/bin/tree-sitter-stack-graphs/test.rs

```diff
@@ -65,45 +80,84 @@ pub struct Command {
     #[clap(long)]
     show_ignored: bool,
 
-    /// Save graph.
-    #[clap(short = 'G')]
-    #[clap(long)]
-    save_graph: bool,
+    /// Save graph for tests matching output mode.
+    /// Takes an optional path specification argument for the output file.
+    /// [default: %n.graph.json]
+    #[clap(
+        long,
+        short = 'G',
+        value_name = "PATH_SPEC",
+        min_values = 0,
+        max_values = 1,
+        require_equals = true,
+        default_missing_value = "%n.graph.json"
+    )]
+    save_graph: Option<PathSpec>,
 
-    /// Save paths.
-    #[clap(short = 'P')]
-    #[clap(long)]
-    save_paths: bool,
+    /// Save paths for tests matching output mode.
+    /// Takes an optional path specification argument for the output file.
+    /// [default: %n.paths.json]
+    #[clap(
+        long,
+        short = 'P',
+        value_name = "PATH_SPEC",
+        min_values = 0,
+        max_values = 1,
+        require_equals = true,
+        default_missing_value = "%n.paths.json"
+    )]
+    save_paths: Option<PathSpec>,
 
-    /// Save visualization.
-    #[clap(short = 'V')]
-    #[clap(long)]
-    save_visualization: bool,
+    /// Save visualization for tests matching output mode.
+    /// Takes an optional path specification argument for the output file.
+    /// [default: %n.html]
+    #[clap(
+        long,
+        short = 'V',
+        value_name = "PATH_SPEC",
+        min_values = 0,
+        max_values = 1,
+        require_equals = true,
+        default_missing_value = "%n.html"
+    )]
+    save_visualization: Option<PathSpec>,
 
     /// Controls when graphs, paths, or visualization are saved.
-    #[clap(long)]
-    #[clap(arg_enum)]
-    #[clap(default_value_t = OutputMode::OnFailure)]
+    #[clap(long, arg_enum, default_value_t = OutputMode::OnFailure)]
     output_mode: OutputMode,
 }
 
+fn validate_path(path: &str) -> Result<PathBuf, String> {
```

Comment:
Thanks for catching that! I have used `Path` already in most places, so that Windows paths should be supported. I'll change this validator to also work on `OsStr` and accepts all valid Windows path.

I'll also add something to the help that explains how path specifiers and non-Unicode paths work together.